src/MLIR/AST.hs

```diff
@@ -76,7 +76,9 @@ data Type =
   deriving Eq
 
 -- TODO(apaszke): Flesh this out
-data Location = UnknownLocation
+data Location =
+    UnknownLocation
+  | FileLineColLocation BS.ByteString UInt UInt
```

Comment:
Are there any other builtin locations that MLIR supports? I'm wondering if we could just shorten in to `FileLocation`, because this is a little verbose. Also might be nice to use a record instead of a regular ADT to indicate which UInt is what. So for example it would change to
```hs
FileLocation { locPath :: BS.ByteString, locLine :: UInt, locColumn :: UInt }
```

Reply:
Also it's fine to leave the other name in `Native`. That one is supposed to be a closer mirror of the C API, but this module here is meant to be a convenient AST for working with it in Haskell

Reply:
Indeed, there are CallSiteLoc, FusedLoc, NameLoc and OpaqueLoc. Location in effectively a wrapper around LocationAttr and I was wondering about changing this to match that (having LocationAttr below in attributes, have the cases there and this is then a wrapper/patterns for those) but punted on that. But I'll add the other ones here, they aren't exposed via C API yet, so I may just let them fail on export and go add them C API side.

Reply:
Makes sense, thanks.